From d5d0c4bb7cc9aa4a132ec0bea13255aee50c1cf9 Mon Sep 17 00:00:00 2001
From: Maxime Hadjinlian <maxime.hadjinlian@devialet.com>
Date: Fri, 6 Jun 2014 14:55:58 +0200
Subject: [PATCH] Don't link icudata as a data only library

This patch cames straight from Debian. It fixes an issue when
libicudata would not have some flags indicating it's EABIhf, causing
applications linked against libicudata to not start on EABIhf
systems. Getting rid of the -nodefaultlibs -nostdlib flags solves the
problem, and is the solution that is used by Debian, see
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=653457.

Signed-off-by: Maxime Hadjinlian <maxime.hadjinlian@devialet.com>
---
 source/config/mh-linux | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/source/config/mh-linux b/source/config/mh-linux
index 531a3b2..5a2a7c4 100644
--- a/source/config/mh-linux
+++ b/source/config/mh-linux
@@ -21,7 +21,9 @@ LD_RPATH= -Wl,-zorigin,-rpath,'$$'ORIGIN
 LD_RPATH_PRE = -Wl,-rpath,

 ## These are the library specific LDFLAGS
-LDFLAGSICUDT=-nodefaultlibs -nostdlib
+#LDFLAGSICUDT=-nodefaultlibs -nostdlib
+# Debian change: linking icudata as data only causes too many problems.
+LDFLAGSICUDT=

 ## Compiler switch to embed a library name
 # The initial tab in the next line is to prevent icu-config from reading it.
--
2.0.0.rc2

